src/SurveyService.Validation/Option/CreateOptionRequestValidator.cs

```diff
@@ -0,0 +1,42 @@
+﻿using FluentValidation;
+using LT.DigitalOffice.SurveyService.Data.Interfaces;
+using LT.DigitalOffice.SurveyService.Models.Dto.Requests.Option;
+using LT.DigitalOffice.SurveyService.Validation.Option.Interfaces;
+
+namespace LT.DigitalOffice.SurveyService.Validation.Option;
+
+public class CreateOptionRequestValidator : AbstractValidator<(CreateOptionRequest request, bool? hasCustomOptions)>, ICreateOptionRequestValidator
+{
+  private readonly IQuestionRepository _questionRepository;
+
+  public CreateOptionRequestValidator(
+    IQuestionRepository questionRepository)
+  {
+    _questionRepository = questionRepository;
+
+    When(x => x.hasCustomOptions is null, () =>
+    {
+      RuleFor(x => x.request.QuestionId)
+        .MustAsync(async (x, _) => await _questionRepository.GetAsync(x) is not null)
+        .WithMessage("The question id doesn't exist.");
+
+      When(x => x.request.IsCustom, () =>
+      {
+        RuleFor(x => x.request.QuestionId)
+          .MustAsync(async (x, _) => (await _questionRepository.GetAsync(x)).HasCustomOptions)
+          .WithMessage("This question can't have custom options.");
+      });
+    });
+
+    When(x => x.hasCustomOptions is not null && x.request.IsCustom, () =>
+    {
+      RuleFor(x => x.hasCustomOptions)
+        .Must(x => (bool)x)
```

Comment:
x is bool? . Possible InvalidOperationException.

Reply:
I mean x is nullable

Reply:
Lev, there is null-check in When (line 31)

Maya, what do you want to check here? I don't understand

Reply:
> Lev, there is null-check in When (line 31)
Ok, got it

Reply:
Here I check if the question can have custom options, if not, then this is an error